### PR TITLE
Update language on Semtech versions

### PR DESCRIPTION
**Pull request by tekstrand**
https://discord.com/channels/867578229534359593/871540428510474302/1313402465281904650

## What did you change
Updated language recently changed in https://github.com/meshtastic/meshtastic/pull/1613 after further conversation in the docs channel and on the PR

## Why did you change it
The wording added was determined to be inaccurate.

### CODE REVIEW

**Comment by GUVWAF** [comment]
In my opinion you don't need to recommend "avoiding" SX127x radios. Normally you don't directly choose between a SX126x and SX127x, but you choose between different devices, and a SX127x would just be a downside. 

There are some exceptions like the T-Beam and LoRa T3S3 which have different variants, so maybe change it to "If you have a choice, we recommend a SX126x over a SX127x radio."

File: docs/hardware/devices/index.mdx [hunk 1]
```diff
@@ -24,7 +24,7 @@ Please do your research and choose the board that meets your needs (or maybe alr
 
 :::info
 
-- Devices using the previous-generation Semtech SX1276 are no longer recommended due to their inferior performance.
+- We recommend devices that use the newer Semtech SX1262 due to improved performance. Avoid the previous-generation Semtech SX1276.
```
Comment (pdxlocations) [inline docs/hardware/devices/index.mdx:27]:
```suggestion
- If possible, we recommend choosing devices equipped with the newer Semtech SX1262, as it offers improved performance compared to the SX1276.
```

Comment (GUVWAF) [reply docs/hardware/devices/index.mdx:27]: Maybe also change "SX1262" into "SX126x series" and "SX1276" into "SX127x series".

Comment (pdxlocations) [reply docs/hardware/devices/index.mdx:27]: updated the suggestion

**Review by pdxlocations — APPROVED** [review]

**Review by GUVWAF — APPROVED** [review]

**Pull request merged**